File: cpp/Kode Program/Appliance.cpp
#include "Electronic.cpp"
// Turunan dari Electronic.
// Tambah atribut spesifik buat alat elektronik biasa:
// tipe (TV, Laptop, dll), garansi, harga, dan stok.
class Appliance : public Electronic {
private:
    string type;
    int warranty;
    double price;
    int stock;

public:
    // Constructor lengkap
    Appliance(int id, string name, string brand, int year,
              string type, int warranty, double price, int stock)
        : Electronic(id, name, brand, year) {
        this->type = type;
        this->warranty = warranty;
        this->price = price;
        this->stock = stock;
    }

    // Constructor default
    Appliance() : Electronic() {
        this->type = "";
        this->warranty = 0;
        this->price = 0.0;
        this->stock = 0;
    }

    // Getter
    string getType() { return type; }
    int getWarranty() { return warranty; }
    double getPrice() { return price; }
    int getStock() { return stock; }

    // Setter
    void setType(string t) { type = t; }
    void setWarranty(int w) { warranty = w; }
    void setPrice(double p) { price = p; }
    void setStock(int s) { stock = s; }
};

File: python/Kode Program/Appliance.py
from Electronic import Electronic

# Class Appliance
# Turunan dari Electronic.
class Appliance(Electronic):
    def __init__(self, id=0, name="", brand="", year=0,
                 type_="", warranty=0, price=0.0, stock=0):
        super().__init__(id, name, brand, year)
        self.type = type_
        self.warranty = warranty
        self.price = price
        self.stock = stock

    # Getter
    def get_type(self): return self.type
    def get_warranty(self): return self.warranty
    def get_price(self): return self.price
    def get_stock(self): return self.stock

    # Setter
    def set_type(self, t): self.type = t
    def set_warranty(self, w): self.warranty = w
    def set_price(self, p): self.price = p
    def set_stock(self, s): self.stock = s
